src/psyclone/domain/lfric/kern_call_acc_arg_list.py

```diff
@@ -113,6 +113,57 @@ def stencil(self, arg, var_accesses=None):
         var_name = DynStencils.dofmap_name(self._kern.root.symbol_table, arg)
         self.append(var_name, var_accesses)
 
+    def stencil_2d(self, arg, var_accesses=None):
+        '''Add general 2D stencil information associated with the argument
+        'arg' to the argument list. OpenACC requires the full dofmap to be
+        specified. If supplied it also stores this access in var_accesses.This
+        method passes through to the stencil method.
+
+        :param arg: the meta-data description of the kernel \
+            argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        self.stencil(arg, var_accesses)
+
+    def stencil_unknown_extent(self, arg, var_accesses=None):
+        '''Add stencil information to the argument list associated with the
+        argument 'arg' if the extent is unknown. If supplied it also stores
+        this access in var_accesses.
+
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        # The extent is not specified in the metadata so pass the value in
+        from psyclone.dynamo0p3 import DynStencils
```

Comment:
I tried to do this but the tests all fall over with so many errors I can't work out what is going on. As soon as i put the import for DynStencils at the top of the file it seems to break eveything.

Reply:
That'll be down to circular dependencies. In which case, please just put a comment here to say that it has to be here to avoid circular dependencies and disable the pylint warning (search for pylint in the source to see how).